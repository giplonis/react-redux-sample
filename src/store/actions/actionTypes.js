export const FETCH_SUCCESS = 'FETCH_SUCCESS';
export const SELECT_OPTION = 'SELECT_OPTION';
export const FETCH_START = 'FETCH_START';
