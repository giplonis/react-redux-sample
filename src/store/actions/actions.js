import * as types from './actionTypes';
import data from '../data/response.json';

export const fetchData = () => {
  return dispatch => {
    dispatch({
      type: types.FETCH_START,
    });
    setTimeout(() => dispatch(fetchSuccess()), 1000);
  };
};

const fetchSuccess = () => {
  return {
    type: types.FETCH_SUCCESS,
    payload: data,
  };
};

export const selectOption = value => {
  return {
    type: types.SELECT_OPTION,
    value,
  };
};
