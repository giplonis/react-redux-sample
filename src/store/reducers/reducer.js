import * as types from '../actions/actionTypes';

const initialState = {
  payload: null,
  current: 'f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36',
  loading: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case types.FETCH_SUCCESS:
      return { ...state, payload: action.payload, loading: false };
    case types.SELECT_OPTION:
      return {
        ...state,
        current: action.value,
      };
    case types.FETCH_START:
      return {
        ...state,
        loading: true,
      };
    default:
      return state;
  }
};

export default reducer;
