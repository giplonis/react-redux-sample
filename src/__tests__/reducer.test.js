import * as types from '../store/actions/actionTypes';
import reducer from '../store/reducers';

import data from '../store/data/response.json';

describe('store/reducer', () => {
  it('should return initial state when no action is passed', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle FETCH_SUCCESS', () => {
    let state = {
      payload: null,
      current: 'f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36',
      loading: false,
    };
    state = reducer(state, { type: types.FETCH_SUCCESS, payload: data });
    expect(state).toMatchSnapshot();
  });

  it('should handle SELECT_OPTION', () => {
    let state = {
      payload: null,
      current: 'f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36',
      loading: false,
    };
    state = reducer(state, { type: types.SELECT_OPTION, value: 'mockId' });
    expect(state).toMatchSnapshot();
  });
});
