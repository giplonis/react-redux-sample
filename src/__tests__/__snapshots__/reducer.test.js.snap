// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/reducer should handle FETCH_SUCCESS 1`] = `
Object {
  "current": "f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36",
  "loading": false,
  "payload": Object {
    "items": 4,
    "payload": Array [
      Object {
        "id": "f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36",
        "title": "Canceled",
      },
      Object {
        "id": "f234235252352bfc-11233333333e7-a902352327-b2970b7c7222b36",
        "title": "Denied",
      },
      Object {
        "id": "f20863255236c4-2b235235fc-11e3255557-aa20-b2970b3257c7b36",
        "title": "Delayed",
      },
      Object {
        "id": "f20863255236c4-11e5555557-a902352327-aa20-b2970b3257c7b36",
        "title": "Success",
      },
    ],
    "totalItems": 4,
  },
}
`;

exports[`store/reducer should handle SELECT_OPTION 1`] = `
Object {
  "current": "mockId",
  "loading": false,
  "payload": null,
}
`;

exports[`store/reducer should return initial state when no action is passed 1`] = `
Object {
  "current": "f20811144fa-2bf33c-11e5555557-932553252a-b2970b72321c7b36",
  "loading": false,
  "payload": null,
}
`;
